internal/dag/httpproxy_processor.go

```diff
@@ -265,11 +265,16 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 
 				timeout, err := timeout.Parse(auth.ResponseTimeout)
 				if err != nil {
-					validCond.AddErrorf("AuthError", "AuthReponseTimeoutInvalid",
+					validCond.AddErrorf("AuthError", "AuthResponseTimeoutInvalid",
 						"Spec.Virtualhost.Authorization.ResponseTimeout is invalid: %s", err)
 					return
 				}
-				svhost.AuthorizationResponseTimeout = timeout
+
+				if timeout.UseDefault() {
```

Comment:
Maybe collapse this into the pattern we've used before which is UseDefaultOrValue? Wondering if that might read better. Alternatively, a good place is to set the default when the extensionservice is created and have the default be there.

Reply:
Sorry I didn't see this comment when I merged :-(

Maybe something like:
```Go
func (s Setting) DurationOr(t time.Duration) time.Duration {
    if s.UseDefault() {
        return t
    }

    return s.val
}

```

One consideration is that the natural way to do this for a `timeout.Setting` is probably to specify a backup `time.Duration`, but here we have a backup `timeout.Setting`.

Maybe we could add something like:
```Go
package timeout

func FirstOf(settings ...Setting) Setting {
    for _, s := range settings {
        if !s.UseDefault() {
            return s
        }
    }

    return DefaultSetting()
}
```